### ISSUE TITLE
Why access policies are useful even after integrity is achieved

### ISSUE THREAD

**Issue by t2ym** [t2ym/thin-hook#321]
## Why access policies are useful even after integrity is achieved

- Functional issues and vulnerabilities come from software development phases, not from users and attackers
- They are inherent and stealth until users/testers or attackers find them
- Code and data integrity is totally ineffective to avoid them
- Access policies define a dedicated sandbox for each execution context of the code
- To err is human. It is practically impossible to eliminate all issues and vulnerabilities from complex software
- In other words, software cannot fully meet human expectation
- Access policies can detect and avoid violation of the expectation
- The policies are more explicit descriptions of the expectation than the raw code
- It is easier to control issues and vulnerabilities of the policies themselves than those in the raw code
- Such policies can serve as safety nets for inherent issues and vulnerabilities
- Therefore, access policies can provide software with extra robustness and reliability
